python/ray/data/tests/test_execution_optimizer.py

```diff
@@ -576,6 +576,24 @@ def test_aggregate_e2e(
         assert row.as_pydict() == {"value": idx, "count()": 1}
 
 
+def test_streaming_executor(
+    ray_start_regular_shared,
+    enable_optimizer,
+    enable_streaming_executor,
+):
+    ds = ray.data.range(100, parallelism=4)
+    ds = ds.map_batches(lambda x: x)
+    ds = ds.filter(lambda x: x > 0)
+    ds = ds.random_shuffle()
+    ds = ds.map_batches(lambda x: x)
+
+    result = []
+    for batch in ds.iter_batches(batch_size=3):
+        assert len(batch) == 3, batch
+        result.extend(batch)
+    assert sorted(result) == list(range(1, 100)), result
```

Comment:
Nit: Is there a way we could assert that the `StreamingExecutor` is used? Not sure if that's easy to do without inspecting stdout, no worries if it's a hassle.

Reply:
Was thinking about it, and It looks to me there's no easy way, but let me know if otherwise cc @jianoaix.

Reply:
Maybe asserting in the constructor of BulkExecutor and StreamingExecutor based on if the flag in context is on? That may help catch issues like https://github.com/ray-project/ray/issues/32586.

Reply:
I guess we can do it in a follow up, right now the features are not fully complete, so assertion would fail in too many places.

Reply:
Yeah SG to followup.